### DISCUSSION TITLE
Upgrade to .NET 7 / API8

### DISCUSSION THREAD

**Discussion by goaaats** [goatcorp/Dalamud#1042] [Announcements]
The .NET 7 branch is now available for testing. You can switch to it with the branch switcher in the dev menu or with kind `net7` and key `tangerine`.

## Changes
Please see the list of breaking changes in the runtime, most of which will likely not affect you: https://learn.microsoft.com/en-us/dotnet/core/compatibility/7.0

### Dalamud Changes
* Most recent FFXIVClientStructs, updated to .NET 6
* Dev plugins that throw while unloading can now be reloaded, though they may be in an inconsistent state - this is best effort
* The ClientState.Login event now only fires as soon as a local player character game object is present
* Newtonsoft.Json updated to 13.0.2
* Unobserved exceptions in tasks will no longer force the user to close the game, they will only be logged
* Non-DIP17 plugins will be going away (only applies to official repo devs)

No other major API changes, but we are planning some bigger stuff for the next cycle.

## Release timeline
We will bump for Patch 6.3 on 09/01.

### Plugin upgrade path
If you want to upgrade your plugins, please follow these steps:
* Change the target framework to net7.0 or net7.0-windows
* Upgrade all of your System.* dependencies to 7.0.0
* Upgrade to the latest DalamudPackager
* Submit them once you tested them on the patched version of the game

DIP17 builds are already done with the .NET7 SDK and pick the correct runtime.

## Reporting issues
Please leave a comment here if you encounter any issues.

**Comment by Pilzinsel64** [comment]
Switched to net7 few hours ago. It works like a charm. No chrashes, no errors, all my ~40 plugins works fine without problem.
If I get an error or something else, I will tell you. :)

**Comment by aers** [comment]
ClientStructs should be .net7 too around tomorrow, barring some major setback.
